CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,162 @@
+or unlogged forums such as IRC channel history, without the sender's consent.
+- Personal insults, especially those using racist or sexist terms.
+- Unwelcome sexual attention.
+- Excessive profanity. Please avoid swearwords; people differ greatly in their sensitivity to swearing.
+- Repeated harassment of others. In general, if someone asks you to stop, then stop.
+- Advocating for, or encouraging, any of the above behaviour.
+
+
+Diversity Statement
+-------------------
+
+The PySAL project welcomes and encourages participation by everyone. We are
+committed to being a community that everyone enjoys being part of. Although
+we may not always be able to accommodate each individual's preferences, we try
+our best to treat everyone kindly.
+
+No matter how you identify yourself or how others perceive you: we welcome you.
+Though no list can hope to be comprehensive, we explicitly honour diversity in:
+age, culture, ethnicity, genotype, gender identity or expression, language,
+national origin, neurotype, phenotype, political beliefs, profession, race,
+religion, sexual orientation, socioeconomic status, subculture and technical
+ability, to the extent that these do not conflict with this code of conduct.
+
+
+Though we welcome people fluent in all languages, PySAL development is
+conducted in English.
+
+Standards for behaviour in the PySAL community are detailed in the Code of
+Conduct above. Participants in our community should uphold these standards
+in all their interactions and help others to do so as well (see next section).
+
+
+Reporting Guidelines
+--------------------
+
+We know that it is painfully common for internet communication to start at or
+devolve into obvious and flagrant abuse.  We also recognize that sometimes
+people may have a bad day, or be unaware of some of the guidelines in this Code
+of Conduct. Please keep this in mind when deciding on how to respond to a
+breach of this Code.
+
+For clearly intentional breaches, report those to the Code of Conduct committee
+(see below). For possibly unintentional breaches, you may reply to the person
+and point out this code of conduct (either in public or in private, whatever is
+most appropriate). If you would prefer not to do that, please feel free to
+report to the Code of Conduct Committee directly, or ask the Committee for
+advice, in confidence.
+
+You can report issues to the PySAL Code of Conduct committee, at
+***pysal-conduct@googlegroups.com***. Currently, the committee consists of:
+
+- ***pysaler 1***
+- ***pysaler 2***
+- ***James Gaboardi***
+
+If your report involves any members of the committee, or if they feel they have
+a conflict of interest in handling it, then they will recuse themselves from
+considering your report. Alternatively, if for any reason you feel
+uncomfortable making a report to the committee, then you can also contact:
+
+- ***somebody other than pysaler 1, 2, or James***
```

Comment:
need an outsider

Reply:
tough one... I'd be tempted to say that you can always go to @sjsrey, but that's not quite in the spirit of the original document. We might need to think about a good external.

Reply:
I thought [this article](https://opensource.guide/code-of-conduct/) on designing a code of conduct was informative. It is not clear if these policies _require_ an "outside" person. It would be great to have numfocus as an outside participant in pysal's CoC. However, pysal's size and numfocus's other duties may not necessitate/allow for this. I think listing @sjsrey as the committee backup is reasonable, I would also be happy to take this role.